aws/code/HPOConfig.py

```diff
@@ -0,0 +1,228 @@
+                'verbosity': 0,
+                'seed': args.seed,
+                'objective': 'binary:logistic'
+            }
+
+            if 'single-CPU' in self.compute_type:
+                model_params.update({'nthreads': os.cpu_count()})
+
+            if 'GPU' in self.compute_type:
+                model_params.update({'tree_method': 'gpu_hist'})
+            else:
+                model_params.update({'tree_method': 'hist'})
+
+        elif 'RandomForest' in self.model_type:
+            # intentionally breaking PEP8 below for argument alignment
+            parser.add_argument( '--max_depth'   , type = int,   default = 5)
+            parser.add_argument( '--n_estimators', type = int,   default = 10)
+            parser.add_argument( '--max_features', type = float, default = 1.0)
+            parser.add_argument( '--n_bins'      , type = float, default = 64)
+            parser.add_argument( '--bootstrap'   , type = bool,  default = True)
+            parser.add_argument( '--random_state', type = int,   default = 0)
+
+            args, unknown_args = parser.parse_known_args(input_args)
+
+            model_params = {
+                'max_depth': args.max_depth,
+                'n_estimators': args.n_estimators,
+                'max_features': args.max_features,
+                'n_bins': args.n_bins,
+                'bootstrap': args.bootstrap,
+                'random_state': args.random_state
+            }
+
+        else:
+            raise Exception(f"!error: unknown model type {self.model_type}")
+
+        pprint.pprint(model_params, indent=5)
+        return model_params
+
+    def detect_data_inputs(self, directory_structure):
+        """
+        Scan mounted data directory to determine files to ingest.
+        Notes: single-CPU pandas read_parquet needs a directory input
+               single-GPU cudf read_parquet needs a list of files
+               multi-CPU/GPU can accept either a list or a directory
+        """
+        parquet_files = glob.glob(
+            os.path.join(directory_structure['train_data'], '*.parquet')
+        )
+        csv_files = glob.glob(
+            os.path.join(directory_structure['train_data'], '*.csv')
+        )
+
+        if len(csv_files):
+            hpo_log.info('CSV input files detected')
+            target_files = csv_files
+            input_file_type = 'CSV'
+
+        elif len(parquet_files):
+            hpo_log.info('Parquet input files detected')
+            if 'single-CPU' in self.compute_type:
+                # pandas read_parquet needs a directory input
+                target_files = directory_structure['train_data'] + '/'
+            else:
+                target_files = parquet_files
+            input_file_type = 'Parquet'
+        else:
+            raise Exception("! No [CSV or Parquet] input files detected")
+
+        n_datafiles = len(target_files)
+        assert (n_datafiles > 0)
+
+        pprint.pprint(target_files)
```

Comment:
If you want to move this over to a logging function, [pformat ](https://docs.python.org/3/library/pprint.html#pprint.pformat)will do what you need.

Reply:
Since we used `pformat` at line 173, do we want to do the same here? Don't mind either way, so long as it's consistent.